tests/acceptance/crate-test.js

```diff
@@ -125,9 +126,41 @@ test('crates license is supplied by version', async function(assert) {
     assert.dom('.license').hasText('MIT/Apache-2.0');
 });
 
+test('navigating to the owners page when not logged in', async function(assert) {
+    server.loadFixtures();
+
+    await visit('/crates/nanomsg');
+
+    assert.dom('#crate-owners p a').doesNotExist();
+});
+
+test('navigating to the owners page when not an owner', async function(assert) {
+    server.loadFixtures();
+
+    this.application.register('service:session-b', Ember.Service.extend({
+        currentUser: {
+            id: 'iain8'
+        }
+    }));
+
+    this.application.inject('controller', 'session', 'service:session-b');
+
+    await visit('/crates/nanomsg');
+
+    assert.dom('#crate-owners p a').doesNotExist();
+});
+
 test('navigating to the owners page', async function(assert) {
     server.loadFixtures();
 
+    this.application.register('service:session-b', Ember.Service.extend({
+        currentUser: {
+            id: 'thehydroimpulse'
+        }
+    }));
+
+    this.application.inject('controller', 'session', 'service:session-b');
```

Comment:
OMG COOL THIS IS HOW TO FAKE LOGGING IN IN THE JAVASCRIPT TESTS!! I tried to figure this out for a long time and couldn't!!!

Reply:
yes! 😄 it is very unclear in the documentation that you have to do this and can't use an existing service.